### PR TITLE
Completed Reset Password Page

### LINKED ISSUE #4: Reset Password Page
Forgot Password Flow
1. Login Page (completed)
2. Forgot Password Password Page (completed)
3. Check Email for Unique Link to Reset Page (handled by portal API)
4. **Open Reset Password Page** (unique page link can only by found from the email they received which lets us know which user account it's for without further authorization
5. **Fill Out Page**
6. **Submit** (redirects to login page)

Need to complete steps 4-6. 

UI should be fairly simple following our existing auth components with an onSubmit that redirects to the login page after making the API call.

Screenshot 2023-04-04 at 5 39 04 AM

### PR DESCRIPTION
**Pull request by farisashai**
- Completed reset password page to read access code from URL param and post to API with form data containing new password.
- Modified react-hook-form onSubmit type slightly for simplicity.
- Restyled register page to take up less vertical space

Closes #4 

I don't think it's possible to test a valid use case of this page in Cypress since we need to fetch the access code from a genuine email sent to the user so I've included screenshots of every edge case.

### Failing Validation on Both Fields
Screenshot 2023-04-07 at 1 07 01 AM

### Mismatched Passwords
Screenshot 2023-04-07 at 1 07 17 AM

### Valid Usage Response
 Redirect to login with toast
Screenshot 2023-04-07 at 1 07 25 AM

### Invalid URL param accessCode
Screenshot 2023-04-07 at 1 10 22 AM

### CODE REVIEW

File: src/pages/reset-password/[accessCode].tsx [hunk 1]
```diff
@@ -0,0 +1,103 @@
...
+  return (
+    <VerticalForm onEnterPress={handleSubmit(onSubmit)}>
+      <SignInTitle text="Reset Password" />
+      <SignInFormItem
+        icon={<VscLock />}
+        type="password"
+        element="input"
+        name="password"
+        placeholder="Password"
+        formRegister={register('newPassword', {
```
Comment (trevorkw7) [inline src/pages/reset-password/[accessCode].tsx:56]: The validation logic in both `newPassword` and `confirmPassword` seem to be the same so it might make sense to create a custom validation function defined outside the component that can be used for both fields and any future password validation input boxes (create new account page for example).

**Review by trevorkw7 — COMMENTED** [review]
Other than that the code seems functional and well-structured!

File: src/pages/reset-password/[accessCode].tsx (hunk 1 above)
Comment (farisashai) [reply src/pages/reset-password/[accessCode].tsx:56]: I refactored the code for validation logic across all of the existing forms we have so far and changed a decent bit of code that is unrelated so the Reset Password page so I'll open a separate PR with the changes to allow for a more isolated conversation for the larger refactor.

Comment (farisashai) [reply src/pages/reset-password/[accessCode].tsx:56]: https://github.com/acmucsd/membership-portal-ui-v2/pull/17

**Review by alexzhang1618 — APPROVED** [review]
Changes look good to me 🤔
